### PR TITLE
Hack: Make hack icon removable

### PR DESCRIPTION
**Pull request by danigm**
https://phabricator.endlessm.com/T27905

### CODE REVIEW

File: js/ui/shellDBus.js [hunk 1]
```diff
@@ -501,6 +507,12 @@ var AppStoreService = class {
     }
 
     RemoveApplication(id) {
+        if (appId === 'com.hack_computer.Clubhouse') {
```
[the remark this answers is not in the document]
Comment (danigm) [reply js/ui/shellDBus.js]: Yes, you're right, we should use the `id`.

File: js/ui/appDisplay.js [hunk 2]
```diff
@@ -2882,6 +2887,11 @@ class HackAppIcon extends AppIcon {
         return 'com.hack_computer.Clubhouse';
     }
 
+    remove() {
+        global.settings.set_boolean('show-hack-launcher', false);
```
[the remark this answers is not in the document]
Comment (danigm) [reply js/ui/appDisplay.js]: done

File: js/ui/shellDBus.js [hunk 3]
```diff
@@ -501,6 +507,12 @@ var AppStoreService = class {
...
+            global.settings.set_boolean('show-hack-launcher', false);
```
[the remark this answers is not in the document]
Comment (danigm) [reply js/ui/shellDBus.js]: this should be changed before the IconGridLayout update to reflect the change because.

**Comment by danigm** [comment]
I've just override the `animateLaunch` as @cfoch suggested to make it consistent, so we don't have animations on launch.

**Comment by danigm** [comment]
I've just rebased this PR on top on the hack rebase branch

File: data/org.gnome.shell.gschema.xml.in [hunk 4]
```diff
@@ -158,6 +158,11 @@
       <summary>Pulse Hack desktop icon</summary>
       <description>If true, the Hack desktop icon will pulse to attract the user.</description>
     </key>
+    <key type="b" name="show-hack-launcher">
```
Comment (wjt) [inline data/org.gnome.shell.gschema.xml.in]: I think we should avoid putting any more Endless-specific GSettings in the `org.gnome.shell` namespace, though I accept it's unlikely that upstream will use `show-hack-launcher`.

File: data/org.gnome.shell.gschema.xml.in [hunk 5]
```diff
@@ -158,6 +158,11 @@
...
+      <default>true</default>
+      <summary>Show the hack launcher in the desktop</summary>
```
Comment (wjt) [inline data/org.gnome.shell.gschema.xml.in]:
```suggestion
      <summary>Show the Hack launcher in the desktop</summary>
```

**Pull request merged**